presto-native-execution/presto_cpp/main/operators/BroadcastFactory.cpp

```diff
@@ -146,10 +146,16 @@ velox::BufferPtr BroadcastFileReader::next() {
 
   auto readFile = fileSystem_->openFileForRead(
       broadcastFileInfos_[readfileIndex_].filePath_);
-  auto buffer = AlignedBuffer::allocate<char>(readFile->size(), pool_, 0);
-  readFile->pread(0, readFile->size(), buffer->asMutable<char>());
   ++readfileIndex_;
   ++numFiles_;
+
+  // Empty broadcast file.
```

Comment:
Why would we produce an empty broadcast file in the first place?

Reply:
This gets generated when we have empty vectors. We can add handling to track how many rows have been written and return empty pages/no page for those scenarios.

Reply:
> We can add handling to track how many rows have been written and return empty pages/no page for those scenarios.

That would be better I think. Thanks.

Reply:
Updated code to not open file till we have non zero rows to write.